src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -858,7 +862,31 @@ namespace cryptonote
 
         uint64_t previous_height = m_core.get_current_blockchain_height();
 
+        // we lock all the rest to avoid having multiple connections redo a lot
+        // of the same work, and one of them doing it for nothing: subsequent
+        // connections will wait until the current one's added its blocks, then
+        // will add any extra it has, if any
+        CRITICAL_REGION_LOCAL(m_sync_lock);
```

Comment:
Quite possibly. The pause/unpause uses a counter AFAIK, so it's resistant to "random" ordering from multiple threads, if that's what you were saying. I'll check it out soon.